git/client.go

```diff
@@ -348,6 +350,22 @@ func (c *Client) DeleteLocalBranch(ctx context.Context, branch string) error {
 	return nil
 }
 
+func (c *Client) DeleteRemoteBranch(ctx context.Context, client *api.Client, repo ghrepo.Interface, branch string) error {
+	if isLocal, err := c.IsLocalGitRepo(ctx); err != nil || !isLocal {
+		return api.BranchDeleteRemote(client, repo, branch)
+	}
+	args := []string{"push", "--delete", "origin", branch}
+	cmd, err := c.Command(ctx, args...)
+	if err != nil {
+		return err
+	}
+	_, err = cmd.Output()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
```

Comment:
I believe I understand the motivation here _(ensure we delete the branch whether it is local or remote)_, however my knee jerk is to keep the git client library purely focused on managing `git` repository locally.  In this case, the reasoning of whether delete the remote branch should live within the `cmd`.

Reply:
Since this needs to be used for `pr close` and `pr merge`, perhaps `pkg/cmg/pr/shared` is a better home for this helper 🤔 allowing the git client package to focus on git with a reusable helper around multiple `gh pr` commands.

Reply:
okay, that seems like an easy place to migrate all the helper code